src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs

```diff
@@ -1130,8 +1132,42 @@ public TBServer()
             format.WaitForExit();
             if (format.ExitCode != 0) throw new InvalidOperationException("format failed");
 
-            process = Process.Start(TB_SERVER, START);
+            process = new Process();
+            process.StartInfo.FileName = TB_SERVER;
+            process.StartInfo.Arguments = START;
+            process.StartInfo.RedirectStandardError = true;
+            process.Start();
             if (process.WaitForExit(100)) throw new InvalidOperationException("Tigerbeetle server failed to start");
+
+            var readPortEvent = new AutoResetEvent(false);
+            var addressLogged = "";
+            var stderrLogged = new StringBuilder();
+            process.ErrorDataReceived += (sender, e) =>
+            {
+                var line = e.Data;
+                if (line == null) return;
+                stderrLogged.Append(line);
+                var listening = "listening on ";
+                var found = line.IndexOf(listening);
+                if (found != -1)
+                {
+                    if (addressLogged != "") throw new InvalidOperationException("have already read the port");
+                    addressLogged = line.Substring(found + listening.Length).Trim();
+                    readPortEvent.Set();
+                }
+            };
+            process.BeginErrorReadLine();

```

Comment:
> https://github.com/tigerbeetle/tigerbeetle/actions/runs/6498235776/job/17649591312?pr=1221

It's bizarre, it couldn't find any "listening on" string on the stderr:

```
System.InvalidOperationException: failed to read the port, exited=False stderr:
info(io): opening "dotnet-tests.tigerbeetle"...warning(main): Grid cache size of 256MB is small. See --cache-grid
```

My guess (and it's only a guess, I haven't debugged), is that the event only fires when the stderr receives a `\n`... being "listening on" the last message, it does not have a line break to fire the event. It's easy to test this hypothesis, by just putting a `\n` on TB's side.

Reply:
We actually do add `\n` here -- `listening on` is a log line, logs go to unbufferred stderr and include `\n`. What's more, I included `.WaitForExit()` in the throwing branch, as, according to the docs, it should collect all the output there is. And there's no output after this line. 

Actually, I think I get it! The next line should read:

```
info(main): 0: Allocated 5099MB in 15 regions during replica init (Grid Cache: 256MB)
```

And we are not getting that! 

I bet we just OOM killed!

Reply:
Makes sense, so try to print the exit code, MacOS always gives 137 when the OOM hits

Reply:
I think I encountered this problem before when the grid cache was updated to 256MB:
https://github.com/tigerbeetle/tigerbeetle/blob/998668b8bcd6ededef09ee28060df76dc8848da6/.github/workflows/clients-java.yml#L273